src/settingsgen/settingsgen.cpp

```diff
@@ -493,15 +490,14 @@ int CDECL main(int argc, char *argv[])
 		CopyFile(after_file, fp);
 		fclose(fp);
 
+		std::error_code error_code;
 		if (CompareFiles(tmp_output, output_file)) {
 			/* Files are equal. tmp2.xxx is not needed. */
-			unlink(tmp_output);
+			std::filesystem::remove(tmp_output, error_code);
```

Comment:
Should we maybe fail when removing the file fails?

Reply:
Wondering if it adds anything to do so?

Reply:
Worst case for not doing it would be potentially causing an error on the next run that the file cannot be overwritten.
On the other hand, should the CI fail due to that? I don't know. That's why I was pondering.

Reply:
I'll leave it out for now, if it is to be checked, the change should probably cover both settingsgen and strgen.